health-checker/src/main/java/io/stargate/health/CheckerResource.java

```diff
@@ -33,12 +45,40 @@ public Response checkLiveness() {
 
   @GET
   @Path("/readiness")
-  public Response checkReadiness() {
-    // Readiness requires a more heavy weight check to decide if the app is able to successfully
-    // respond to traffic
-    if (bundleService.checkIsReady()) {
+  public Response checkReadiness(@QueryParam("check") Set<String> checks) {
+    boolean checkAll = checks == null || checks.isEmpty();
+
+    HashSet<String> requiredChecks =
+        new HashSet<>(checkAll ? bundleService.defaultHeathCheckNames() : checks);
+
+    SortedMap<String, Result> status =
+        healthCheckRegistry.runHealthChecks(
+            (name, healthCheck) -> checkAll || requiredChecks.contains(name));
+
+    boolean started = bundleService.checkBundleStates();
+    boolean dataStoreAvailable = bundleService.checkDataStoreAvailable();
+    boolean ready = started && dataStoreAvailable;
+
+    for (Entry<String, Result> e : status.entrySet()) {
+      String name = e.getKey();
+      requiredChecks.remove(name);
+      boolean healthy = e.getValue().isHealthy();
+
+      if (!healthy) {
+        logger.warn("Failed health check: {}", name);
+        ready = false;
+      }
+    }
+
+    if (!requiredChecks.isEmpty()) {
+      logger.warn("Missing required health checks: {}", requiredChecks);
+      ready = false;
+    }
+
+    if (ready) {
       return Response.status(Response.Status.OK).entity("READY").build();
+    } else {
+      return Response.status(Response.Status.SERVICE_UNAVAILABLE).entity("NOT READY").build();
```

Comment:
I think that we should include a list of unavailable health checks in the entity to give the caller more info. In case the checks lists will have N value, the caller will not know (without looking at the logs) which health checked failed. The UX of this is non-ideal.

Reply:
I thought about that but did not want to alter the API behaviour from the client's perspective.

We also have the `/healthcheck` resource that reports full checker details including the message and timestamp. The only validations this method performs that will be missing from `/healthcheck` output are the bundle states and `DataStore` access. If exposing them to the client is worthwhile, I'd rather add them to `/healthcheck`. WDYT?

Reply:
How does it alter the behavior? We will return proper status, only the entity will contain more data.
I think that ideally, for example for `check=a&check=b` we could return
```
NOT READY
a = HEALTHY
b = UNHELATHY
``` 
That way our API does not change, but we are giving more info to our customers, wdyt?

Reply:
If you're in k8s all you care about is 200 or non-200 responses the response body is mostly ignored/swallowed. So if checks are failing you'd check the logs anyway.

Reply:
Will it be ALWAYS run in the k8s env? If yes, then I agree. However, if there will be some users that will run stargate themself, the additional monitoring information may be useful, right?

Reply:
but we have `/healthcheck` that shows that info. Why would we want to expose the same data in two places?

We can add extra info to `/healthcheck` since it's more of an investigation tool, while `readiness` is supposed to be an on/off quick check. WDYT?

Reply:
sounds good.

Reply:
Implemented in f90f105f241098f9bc651ec256d4bd89cdcbd292.

Sample `/healthcheck` output:

```
{
  "bundles": {
    "healthy": false,
    "message": "Inactive bundles: [io.stargate.db, io.stargate.web]",
    "duration": 0,
    "timestamp": "2021-01-29T12:43:11.298-05:00"
  },
  "cql": {
    "healthy": true,
    "message": "Available",
    "duration": 0,
    "timestamp": "2021-01-29T12:43:11.299-05:00"
  },
  "datastore": {
    "healthy": true,
    "message": "DataStore is operational",
    "duration": 27,
    "timestamp": "2021-01-29T12:43:11.298-05:00"
  },
  "deadlocks": {
    "healthy": true,
    "duration": 0,
    "timestamp": "2021-01-29T12:43:11.299-05:00"
  }
}
```